lib/media/track/localaudiotrack.js

```diff
@@ -20,14 +24,42 @@ const LocalMediaAudioTrack = mixinLocalMediaTrack(AudioTrack);
  * @emits LocalAudioTrack#started
  * @emits LocalAudioTrack#stopped
  */
+
 class LocalAudioTrack extends LocalMediaAudioTrack {
   /**
    * Construct a {@link LocalAudioTrack} from a MediaStreamTrack.
    * @param {MediaStreamTrack} mediaStreamTrack - An audio MediaStreamTrack
    * @param {LocalTrackOptions} [options] - {@link LocalTrack} options
    */
   constructor(mediaStreamTrack, options) {
+    // NOTE(mpatwardhan): by default workaround for WebKitBug1208516 will be enabled on Safari browsers
+    // although the bug is seen  mainly on iOS devices, we do not have a reliable way to tell iOS from MacOs
+    // userAgent on iOS pretends its macOs if Safari is set to request desktop pages.
+    const workaroundWebKitBug1208516 = guessBrowser() === 'safari' &&
+    // eslint-disable-next-line no-undefined
+    document && document.addEventListener && document.visibilityState !== undefined;
```

Comment:
Accessing `document` directly will result in a `TypeError` if its not present. In order to prevent this, we should use `typeof`. Also, it is not good practice to compare with `undefined` directly.
```suggestion
    const workaroundWebKitBug1208516 = guessBrowser() === 'safari'
      && typeof document === 'object'
      && typeof document.addEventListener === 'function'
      && typeof document.visibilityState === 'string';
```